lib/pmdtester/cmd.rb

```diff
@@ -0,0 +1,25 @@
+require 'open3'
+
+module PmdTester
+  class Cmd
+
+    PMD_EXIT_STATUS = 4
+
+    def self.execute(cmd)
+      puts cmd
+
+      stdin, stdout, stderr, wait_thr = Open3.popen3("#{cmd};")
+
+      unless wait_thr.value.success? || wait_thr.value.exitstatus == PMD_EXIT_STATUS
```

Comment:
Do I understand this right? We consider here exit code 0 and exit code 4 as good (for the PmdTester). If PMD exits with any other code, we fail and stop any processing, correct? That is the right behavior.

Maybe rename "PMD_EXIT_STATUS" to "PMD_VIOLATIONS_FOUND".

Reply:
> Do I understand this right?

Yes, you are right. Hm... I have no good idea about how to add test cases for `PmdReportBuilder` now. Maybe we could merge the PR first.

Reply:
I've just merged this PR.

We would ignore the exit code "4" for any command, regardless whether we executed PMD or some other command. This is not so good.

Since we are not interested in failing, when violations are found, we can use the command line option for pmd `-failOnViolation false` (https://pmd.github.io/pmd/pmd_userdocs_getting_started.html#exit-status) when we execute pmd.

About the test - yes. Definitely separate PR.
We could add/commit a small test repo under test resources (e.g. create a git init --bare repo) and use this for the testing.... then we don't depend on external resources for the unit tests.